agents/agents-core/src/commonMain/kotlin/ai/koog/agents/core/agent/context/AIAgentContext.kt

```diff
@@ -31,17 +31,57 @@ import kotlin.uuid.Uuid
  */
 @OptIn(ExperimentalUuidApi::class)
 internal class AIAgentContext(
-    override val environment: AIAgentEnvironment,
-    override val agentInput: String,
-    override val config: AIAgentConfigBase,
-    override val llm: AIAgentLLMContext,
-    override val stateManager: AIAgentStateManager,
-    override val storage: AIAgentStorage,
-    override val sessionUuid: Uuid,
-    override val strategyId: String,
+    environment: AIAgentEnvironment,
+    agentInput: String,
+    config: AIAgentConfigBase,
+    llm: AIAgentLLMContext,
+    stateManager: AIAgentStateManager,
+    storage: AIAgentStorage,
+    sessionUuid: Uuid,
+    strategyId: String,
     @OptIn(InternalAgentsApi::class)
-    override val pipeline: AIAgentPipeline,
+    pipeline: AIAgentPipeline,
 ) : AIAgentContextBase {
+    private var _environment: AIAgentEnvironment = environment
```

Comment:
To make attributes private set, but not sure this is the right way to go. Looks not so ugly as we discussed to make `AIAgentContextDelegate` with `AIAgentContext` context inside